### PR TITLE
19774 Fixed null officer email schema error

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* bcgov/entity#19774

*Description of changes:*
- app version = 5.8.11
- delete officer email if it's null

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

**Review by JazzarKarim — APPROVED** [review]

**Review by vysakh-menon-aot — APPROVED** [review]

File: src/mixins/filing-template-mixin.ts [hunk 1]
```diff
@@ -1468,14 +1468,15 @@ export default class FilingTemplateMixin extends Mixins(AmalgamationMixin, DateM
...
+   * Fixes addresses by deleting the type in case it is null. Also fixes null officer email.
    * @param orgPeople The array of orgs/people
-   * @returns the array of orgs/people after addresses been fixed
+   * @returns the array of orgs/people with fixed address types and email
    */
   fixNullAddressType (orgPeople: OrgPersonIF[]): OrgPersonIF[] {
     return orgPeople.map(p => {
       if (p.deliveryAddress?.addressType === null) delete p.deliveryAddress.addressType
       if (p.mailingAddress?.addressType === null) delete p.mailingAddress.addressType
+      if (p.officer?.email === null) delete p.officer.email
```
Comment (severinbeauvais) [inline src/mixins/filing-template-mixin.ts:1479]: [Party schema](https://github.com/bcgov/business-schemas/blob/main/src/registry_schemas/schemas/party.json) shows this:

image

[JSON schema reference](https://json-schema.org/understanding-json-schema/reference/string#string) shows this:

image

**Comment by severinbeauvais** [comment]
After this change, you can see that parties[0].officer and parties[3].office have an email property but the other 2 don't. This passes schema validation.

image

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-create-dev--pr-651-pp38jdpg.web.app

**Pull request merged**